actionpack/lib/action_dispatch/testing/test_helpers/page_dump_helper.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module ActionDispatch
+  module TestHelpers
+    module PageDumpHelper
+      class InvalidResponse < StandardError; end
+
+      # Saves the content of response body to a file and tries to open it in your browser.
+      # Launchy must be present in your Gemfile for the page to open automatically.
+      def save_and_open_page(path = html_dump_defaul_path)
+        save_page(path).tap { |s_path| open_file(s_path) }
+      end
+
+      private
+        def save_page(path = html_dump_defaul_path)
+          raise InvalidResponse.new("Response is a redirection!") if response.redirection?
+          path = Pathname.new(path)
+          path.dirname.mkpath
+          File.write(path, response.body)
+          path
+        end
+
+        def open_file(path)
+          require "launchy"
+          Launchy.open(path)
```

Comment:
I think we would have to make this a direct dependency of actionpack, which we I'm not sure we want to do

Reply:
Thank you for the review.

I brought over the LoadError rescue from Capybara to prevent making it a hard dependency. 

This is not a new pattern in Rails as it is already used to prevent ActiveSupport/ActionPack from depending on :
- [msgpack](https://github.com/rails/rails/blob/7c03e4f1b7553084f34318a9efc2793137de36f4/activesupport/lib/active_support/message_pack.rb#L3-L10)
- [dalli](https://github.com/rails/rails/commit/9c0ab33c2edf0a20769b32c326b09da17c54ae4e#diff-dd77c85d34316439953d8671b2a38a6050795af7c62e2e3cd3d6e73c2466fd61R2-R7)
- [rexml](https://github.com/rails/rails/commit/f624ed09abba4a0bae64eb699c9fc2e3f9bdc2dc#diff-f91df54f9e283543b6c1d20c771839fcd9bb50f110748e9cd7846f8d7f8fa63dR41-R48)
- [redis](https://github.com/rails/rails/blob/7c03e4f1b7553084f34318a9efc2793137de36f4/activesupport/lib/active_support/cache/redis_cache_store.rb#L3-L10)
- [vips](https://github.com/rails/rails/blob/7c03e4f1b7553084f34318a9efc2793137de36f4/activestorage/lib/active_storage/analyzer/image_analyzer/vips.rb#L11-L18)
 
 Would the "soft" dependency still be a problem? 
It would be totally opt-in since it is never loaded unless you manually put it in your Gemfile.